git-sync/main.go

```diff
@@ -66,8 +84,24 @@ func main() {
 	if _, err := exec.LookPath("git"); err != nil {
 		log.Fatalf("required git executable not found: %v", err)
 	}
+
+	if *flUsername != "" && *flPassword != "" {
+		log.Println("creating .netrc file (required for https authentication)")
+		if err := createNetrcFile(*flUsername, *flPassword, *flRepo); err != nil {
+			log.Fatalf("error creating .netrc file: %v", err)
+		}
+	}
+
+	if *flOneTime {
+		if err := syncRepo(*flRepo, *flDest, *flBranch, *flRev, *flDepth); err != nil {
+			log.Fatalf("error syncing repo: %v", err)
+		}
+
+		os.Exit(0)
+	}
+
 	for {
-		if err := syncRepo(*flRepo, *flDest, *flBranch, *flRev); err != nil {
+		if err := syncRepo(*flRepo, *flDest, *flBranch, *flRev, *flDepth); err != nil {
```

Comment:
same with all of these.  Does passing `flRepo`, for example, not work?  Why pointers?